.circleci/config.yml

```diff
@@ -0,0 +1,126 @@
+  install_yq:
+    description: "Install yq from binary"
+    steps:
+      - run:
+          name: Install yq
+          command: |
+            mkdir -p $HOME/yq
+            curl -L https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64 -o $HOME/yq/yq &&\
+              chmod +x $HOME/yq/yq &&\
+              echo 'export PATH=$HOME/yq/yq:$PATH' >> $BASH_ENV
+  install_gh:
+    description: "Install GitHub Cli"
+    steps:
+      - run:
+          name: Install gh
+          command: |
+            curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+            echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+            sudo apt update
+            sudo apt install gh
+  set_sdk_version:
+    description: "Set Daml SDK version environment version"
+    steps:
+      - run:
+          name: Set Daml SDK version
+          command: echo 'export SDK_VERSION=$(yq e '.sdk-version' daml.yaml)' >> $BASH_ENV
+  run_release:
+    description: "Run Release"
+    steps:
+      - run:
+          name: "Get target release version"
+          command: echo 'export VERSION=$(yq e '.version' daml.yaml)' >> $BASH_ENV
+      - run:
+          name: "Validate target release version"
+          command: |
+            if [ `git tag | grep ${VERSION} | wc -l` -eq 0 ]; then
+              echo "New tag detected ${VERSION}. Releasing..."
+            else
+              echo "Tag ${VERSION} already exists; Skipping Release."
+              circleci-agent step halt
+            fi
+      - run:
+          name: "Tag and push to GitHub"
+          command: gh release create v${VERSION} .daml/dist/daml-ctl-${VERSION}.dar --generate-notes
+      - store_artifacts:
+          path: .daml/dist/daml-ctl-${VERSION}.dar
+          destination: daml-ctl-dar-${VERSION}
+
+jobs:
+  build:
+    executor: daml-docker-executor
+    steps:
+      - checkout
+      - install_yq
+      - restore_cache:
+          keys:
+            - daml-ctl-{{ checksum "daml.yaml" }}
+      - set_sdk_version
+      - install_sdk
+      - run:
+          name: Daml build
+          command: daml build
+      - run:
+          name: Daml test
+          command: daml test
+      - save_cache:
+          paths:
+            - .daml
+          key: daml-ctl-{{ checksum "daml.yaml" }}
+  release:
+    executor: daml-docker-executor
+    steps:
+      - checkout
+      - install_yq
+      - install_gh
+      - restore_cache:
+          keys:
+            - daml-ctl-{{ checksum "daml.yaml" }}
+      - run_release
+
+workflows:
+  version: 2
+  daml_ctl_build:
+    jobs:
+      - build:
+          filters:
+            branches:
+              ignore:
+                - master
+  daml_ctl_release:
```

Comment:
Would it make more sense to rename these to something general so we can re-use these in the other repos i.e. cc and finance?

Reply:
I'll update them